packages/core/src/interpreter.ts

```diff
@@ -511,9 +518,13 @@ export class Actor<TLogic extends AnyActorLogic>
     delay: number;
     to?: AnyActorRef;
   }): void {
-    const timerId = this.clock.setTimeout(() => {
-      this.system._relay(this, to ?? this, event as EventFromLogic<TLogic>);
-    }, delay);
+    const timerId = this.system.scheduler.setTimeout(this, {
+      delay,
+      event,
+      id: Math.random().toString(),
+      startedAt: Date.now(),
+      target: to ?? this
+    });
 
     // TODO: consider the rehydration story here
     if (id) {
```

Comment:
TODO for myself: we should always keep those timer ids around - that would help us to fix test187